resources/benchmark-runner.mjs

```diff
@@ -160,88 +106,57 @@ export class BenchmarkRunner {
             window.performance.mark(name);
     }
 
-    // This function ought be as simple as possible. Don't even use Promise.
-    _runTest(suite, test, page, callback)
+    async runMultipleIterations(iterationCount)
     {
         const self = this;
-        const now = window.performance && window.performance.now ? () => window.performance.now() : Date.now;
-
-        self._writeMark(suite.name + '.' + test.name + '-start');
-        let startTime = now();
-        test.run(page);
-        let endTime = now();
-        self._writeMark(suite.name + '.' + test.name + '-sync-end');
-
-        const syncTime = endTime - startTime;
-
-        startTime = now();
-        setTimeout(() => {
-            // Some browsers don't immediately update the layout for paint.
-            // Force the layout here to ensure we're measuring the layout time.
-            const height = self._frame.contentDocument.body.getBoundingClientRect().height;
-            endTime = now();
-            self._frame.contentWindow._unusedHeightValue = height; // Prevent dead code elimination.
-            self._writeMark(suite.name + '.' + test.name + '-async-end');
-            window.requestAnimationFrame(() => {
-                callback(syncTime, endTime - startTime, height);
-            });
-        }, 0);
+        if (this._client && this._client.willStartFirstIteration)
+            this._client.willStartFirstIteration(iterationCount);
+        for (let i= 0; i< iterationCount; i++) {
+            await self._runAllSuites();
+        }
+        if (this._client && this._client.didFinishLastIteration)
+            this._client.didFinishLastIteration();
     }
 
-    step(state) {
-        if (!state) {
-            state = new BenchmarkState(this._suites);
-            this._measuredValues = {tests: {}, total: 0, mean: NaN, geomean: NaN, score: NaN};
-        }
+    async _runAllSuites()
+    {
+        this._measuredValues = {tests: {}, total: 0, mean: NaN, geomean: NaN, score: NaN};
 
-        const suite = state.currentSuite();
-        if (!suite) {
-            this._finalize();
-            return Promise.resolve();
-        }
+        this._removeFrame();
```

Comment:
Why do we need to call this._removeFrame() here??

Reply:
I'm just cargo-culting here, and moved it out of the `_finalize` method.